From 37a0ba7080e4d4599433037be0873a683be299e5 Mon Sep 17 00:00:00 2001
From: Lars-Dominik Braun <lars@6xq.net>
Date: Mon, 15 Feb 2021 14:03:29 +0100
Subject: [PATCH 3/4] Propagated environment from server process to session

This is useful with auth-none=1 and per-user instances.
---
 src/cpp/server/ServerSessionManager.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cpp/server/ServerSessionManager.cpp b/src/cpp/server/ServerSessionManager.cpp
index 6e37578cab..e586f202a0 100644
--- a/src/cpp/server/ServerSessionManager.cpp
+++ b/src/cpp/server/ServerSessionManager.cpp
@@ -129,6 +129,8 @@ core::system::ProcessConfig sessionProcessConfig(
 
    // pass our uid to instruct rsession to limit rpc clients to us and itself
    core::system::Options environment;
+   // Copy existing environment
+   core::system::environment(&environment);
    uid_t uid = core::system::user::currentUserIdentity().userId;
    environment.push_back(std::make_pair(
                            kRStudioLimitRpcClientUid,
-- 
2.26.2

